source/pages/_document.tsx

```diff
@@ -1,14 +1,46 @@
 import Document, { Head, Html, Main, NextScript } from 'next/document';
+import React from 'react';
+import { CardanoNetwork } from '../constants';
+import { environment } from '../environment';
 import { DEFAULT_LOCALE } from '../features/i18n';
 import { cardanoExplorerTheme } from '../styles/theme/theme';
 
+interface IMyWindow extends Window {
+  dataLayer: any[];
+}
+
+declare var window: IMyWindow;
+
 class CardanoExplorerDocument extends Document {
   public render() {
     const lang =
       this.props.__NEXT_DATA__.props?.pageProps?.locale || DEFAULT_LOCALE;
+
+    const injectGA = () => {
+      if (typeof window === 'undefined') {
+        return;
+      }
+      window.dataLayer = window.dataLayer || [];
+      function gtag(param: string, value: any) {
```

Comment:
Im confused by this `gtag` function - it has the same name as the one used in the navigation store but doesn't have the same signature? Also: is this a hack to expose `gtag` as a global function when injecting GA in the script part below? More detailed explanation is required as it is hard to see what's exactly going on 🙏

Reply:
Well yes and no, because this gtag function is exposed so it available on _document component only. Basically gtag is encapsulated across window object and available on every page. This part of the implementation is just to follow typescript/lint so there are no errors.